libs/lib-client/trigger-selector/src/trigger-selector.component.ts

```diff
@@ -0,0 +1,117 @@
+import { Component, OnInit } from '@angular/core';
+import { ModalControl, ModalService } from '@flogo-web/lib-client/modal';
+import { FlogoInstallerComponent } from '@flogo-web/lib-client/contrib-installer';
+import {
+  TriggersService,
+  Trigger,
+  ContributionsService,
+  FLOGO_CONTRIB_TYPE,
+} from '@flogo-web/lib-client/core';
+import { TriggerSchema } from '@flogo-web/core';
+import { LocalSearch } from '@flogo-web/lib-client/search';
+import { Observable } from 'rxjs';
+export interface TriggerMetaData {
+  appId: string;
+}
+
+export interface TriggerSelectorResult {
+  installType: 'existing' | 'installed';
+  triggerSchema: TriggerSchema;
+  trigger?: Trigger;
+}
+
+@Component({
+  selector: 'flogo-trigger-selector',
+  templateUrl: './trigger-selector.component.html',
+  styleUrls: ['./trigger-selector.component.less'],
+})
+export class TriggerSelectorComponent implements OnInit {
+  showExistingTriggers = true;
+  showExistingTriggersTab: boolean;
+  existingTriggersSearcher: LocalSearch<Trigger>;
+  installedTriggersSearcher: LocalSearch<TriggerSchema>;
+
+  private installedTriggers: TriggerSchema[] = [];
+  private appId: string;
+
+  constructor(
+    private control: ModalControl<TriggerMetaData>,
+    private modalService: ModalService,
+    private triggerService: TriggersService,
+    private contribService: ContributionsService
+  ) {
+    this.appId = this.control.data.appId;
+
+    this.existingTriggersSearcher = new LocalSearch({ matchFields: ['name'] });
+
+    this.installedTriggersSearcher = new LocalSearch({ matchFields: ['title'] });
+  }
+
+  ngOnInit() {
+    this.getInstalledTriggers();
+    this.getExistingTriggers();
+  }
+
+  displayExistingTriggers() {
+    this.showExistingTriggers = true;
+  }
+
+  displayInstalledTriggers() {
+    this.showExistingTriggers = false;
+  }
+
+  addExistingTriggerToApp(trigger, installType) {
+    const triggerSchema = this.installedTriggers.find(schema => {
+      return schema.ref === trigger.ref;
+    });
+    this.control.close({
+      triggerSchema,
+      installType,
+      trigger,
+    });
+  }
+
+  addNewTriggerToApp(triggerSchema, installType) {
+    this.control.close({ triggerSchema, installType });
+  }
+
+  closeTriggerSelector() {
+    this.control.close(null);
+  }
+
+  openInstallTriggerWindow() {
+    this.modalService
+      .openModal<void>(FlogoInstallerComponent)
+      .result.subscribe(status => {
+        if (status === 'success') {
+          this.showExistingTriggers = false;
+          this.ngOnInit();
```

Comment:
can we not call `ngOnInit` directly? it is not a common pattern to call angular lifecycle methods by ourselves, also ngOnInit is currently reloading both existingTriggers and installedTriggers I don't think it is necessary to reload existingTriggers